### PR TITLE
chore: npm distribution

### LINKED ISSUE #523: distribution via NPM
Integrate inertia-cli into NPM

### PR DESCRIPTION
**Pull request by kimoantiqe**
:tickets: **Ticket(s)**: Closes #523 

---

## :construction_worker: Changes

- Added .npm folder  (which contains all npm related code)
- Just to summarize, I have created a node package that downloads CLI based on the user's architecture & platform
## :flashlight: Testing Instructions

Assuming npm is installed 
```script
npm install -g test123cli123
```
```script
inertia --version
```
## :wrench: TODO
- Update package.json with correct package name and so on ( I am currently using test123cli123 , mainly for dev purposes)
- Currently the cli version number is hardcoded , change it so that the version number (to be downloaded from the releases page) matches the node modules version
- Add tests

### CODE REVIEW

File: .npm/.gitignore [hunk 1]
```diff
@@ -0,0 +1,30 @@
+# Logs
+logs
+*.log
+npm-debug.log*
```
Comment (bobheadxi) [inline .npm/.gitignore]: we don't need this file do we?

File: .npm/.eslintrc.json [hunk 2]
```diff
@@ -0,0 +1,6 @@
+{
+    "extends": "standard",
+    "rules": {
+        "no-console":0
+    }
+}
```
Comment (bobheadxi) [inline .npm/.eslintrc.json]: minor nitpick, but we should add newline at end of file 😋

File: .npm/.eslintrc.json [hunk 3]
```diff
@@ -0,0 +1,6 @@
+{
+    "extends": "standard",
+    "rules": {
+        "no-console":0
```
Comment (bobheadxi) [inline .npm/.eslintrc.json]:
```suggestion
        "no-console": 0
```

nitpicking

File: .npm/package.json [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+  },
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/ubclaunchpad/inertia.git"
+  },
+  "keywords": [
+    "inertia",
+    "inertia-cli"
+  ],
+  "author": "UBC-Launchpad",
```
Comment (bobheadxi) [inline .npm/package.json]:
```suggestion
  "author": "UBC Launch Pad",
```

File: .npm/package.json [hunk 5]
```diff
@@ -0,0 +1,36 @@
+{
+  "name": "test123cli123",
+  "version": "0.0.93",
+  "description": "Inertia-CLI",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1",
+    "postinstall": "node ./src/install.js",
+    "preuninstall": "node ./src/uninstall.js",
+    "prepublishOnly": "node ./src/copyREADME.js"
```
Comment (bobheadxi) [inline .npm/package.json]: can we rename this `copyAssets.js`?

File: docs_src/index.html.md [hunk 6]
```diff
@@ -68,6 +68,11 @@ please feel free to
 </aside>
 
 # Getting Started
+> Node.js users can install the CLI using [npm](https://www.npmjs.com/get-npm): 
+
+```shell
+npm install -g inertia-cli
+```
```
Comment (bobheadxi) [inline docs_src/index.html.md]: Can we put this after the `brew` instructions?

File: .npm/.gitignore (hunk 1 above)
Comment (kimoantiqe) [reply .npm/.gitignore]: I guess so!

File: .npm/.eslintrc.json (hunk 2 above)
Comment (kimoantiqe) [reply .npm/.eslintrc.json]: Gonna fix that 👍

File: .npm/.eslintrc.json (hunk 3 above)
Comment (kimoantiqe) [reply .npm/.eslintrc.json]: 👍

File: .npm/package.json (hunk 4 above)
Comment (kimoantiqe) [reply .npm/package.json]: 👍

File: .npm/package.json (hunk 5 above)
Comment (kimoantiqe) [reply .npm/package.json]: Assets make more sense  👍  !

File: docs_src/index.html.md (hunk 6 above)
Comment (kimoantiqe) [reply docs_src/index.html.md]: I put it first cuz npm is not platform specific and I think it's more common than the rest, but sure thing I can reorder it 👍

Comment (bobheadxi) [reply docs_src/index.html.md]: Homebrew is [crossplatform as well now](https://brew.sh/2019/02/02/homebrew-2.0.0/) - though now that I thikn about it, if anyone tries to use the [current formula](https://github.com/ubclaunchpad/homebrew-tap/blob/master/inertia.rb) on another platform it'll break horrible 😋 

> Homebrew officially supports Linux and Windows 10 with Windows Subsystem for Linux (WSL).

**Comment by kimoantiqe** [comment]
@bobheadxi  Nothing in particular is blocking, I just wanted to get your opinion on how everything is laid out cuz npm is stupid .. Once I publish I can only bump versions and I want the node_module version to match inertia's version, so users could simply install using the vanilla npm install @version cmd

**Comment by bobheadxi** [comment]
I don't know if there's a way to automate that 🤔 will you be here this weekend? it might be easier to discuss then!

otherwise, i dont mind having it hardcoded for now and we can open another ticket to improve the process.

File: docs_src/index.html.md (hunk 6 above)
Comment (kimoantiqe) [reply docs_src/index.html.md]: That's super new, right? Just a month ago it was MacOS only 😛

**Comment by kimoantiqe** [comment]
> I don't know if there's a way to automate that 🤔 will you be here this weekend? it might be easier to discuss then!
> 
> otherwise, i dont mind having it hardcoded for now and we can open another ticket to improve the process.

Yes, I will be here! Sure thing we can discuss it then!

File: contrib/npm/package.json [hunk 7]
```diff
@@ -0,0 +1,41 @@
+{
+  "name": "inertia-cli",
+  "version": "0.5.1",
```
Comment (bobheadxi) [inline contrib/npm/package.json]:
```suggestion
  "version": "0.5.2",
```

File: contrib/npm/src/utils.js [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+  linux: 'linux',
+  win32: 'windows',
+  freebsd: 'freebsd'
+};
+
+// Inertia options
+const opts = {
+  binPath: './bin',
+  binName: 'inertia',
+  binversion: '0.5.0'
```
Comment (bobheadxi) [inline contrib/npm/src/utils.js]: could you import this from `package.json`?

```js
const package = require('./package.json');
```

**Pull request merged**